// mongodump/Gruntfile.js
/*
 * mongodump
 * https://github.com/rohit/mongo
 *
 * Copyright (c) 2013 Bhushan Goel
 * Licensed under the MIT license.
 */

'use strict';

module.exports = function(grunt) {

  // Project configuration.
  grunt.initConfig({
    jshint: {
      all: [
        'Gruntfile.js',
        'tasks/*.js',
        '<%= nodeunit.tests %>',
      ],
      options: {
        jshintrc: '.jshintrc',
      },
    },

    // Before generating any new files, remove any previously-created files.
    banner_property: 'AWESOME',
    concat: {
      default_options: {
        files: {
          'build/concat.js': ['src/todo.js', 'src/test.js']
        }
      }
},

    // Configuration to be run (and then tested).
    mongodump: {
      main: {
	
      },
      },

      custom_options: {
        options: {
          separator: ': ',
          punctuation: ' !!!',
        },
        files: {
          'tmp/custom_options': ['test/fixtures/testing', 'test/fixtures/123'],
        },
      },
    

    // Unit tests.
    nodeunit: {
      tests: ['test/*_test.js'],
    },

  });


  // Actually load this plugin's task(s).
  grunt.loadTasks('mongodump');

  // These plugins provide necessary tasks.
  grunt.loadNpmTasks('grunt-contrib-jshint');

  grunt.loadNpmTasks('grunt-contrib-nodeunit');

  // Whenever the "test" task is run, first clean the "tmp" dir, then run this
  // plugin's task(s), then test the result.
  grunt.registerTask('test', ['mongodump', 'nodeunit']);

  // By default, lint and run all tests.
  grunt.registerTask('default', ['jshint', 'test','mongodump']);

};
